### ISSUE TITLE
Bind does not preserve function return type

### ISSUE THREAD

**Issue by Jameskmonger** [Microsoft/TypeScript#16796]
**TypeScript Version:**  2.4.0

**Code**

```ts
function foo<T>(provider: () => T): T { return undefined; }
function giveMeTrue(): boolean { return true; }

const okay: boolean = foo(giveMeTrue); // okay
const notOkay: boolean = foo(giveMeTrue.bind(this)); // type '{}' is not assignable to type 'boolean'
```

**Expected behavior:**

Both should be okay - `bind` does not change the return type of the function.

**Actual behavior:**

`bind` changes the function return type to `any`.

**Comment by jesseschalken** [comment]
There relevant line is here: https://github.com/Microsoft/TypeScript/blob/b162097c3c7f5e5174eb70124de8888669775dbd/lib/lib.d.ts#L271

Should it return `this`?

**Comment by kitsonk** [comment]
Dupe of #212

**Comment by Jameskmonger** [comment]
Closed as duplicate of #212
